clients/src/main/java/org/apache/kafka/common/serialization/ExtendedDeserializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.serialization;
+
+import java.util.Map;
+
+import org.apache.kafka.common.header.Headers;
+
+public interface ExtendedDeserializer<T> extends Deserializer<T> {
```

Comment:
@becketqin As I explained here https://github.com/apache/kafka/pull/2814#issuecomment-291880230, we need to update our system tests infrastructure to run with Java 8 before we can make the switch. It may take a bit of time to get that done, so my suggestion was to do what can be done with Java 7 in the initial PR and file a JIRA for the follow-up work once the Java 8 switch happens. That way, we can make progress instead of being blocked.